### PR TITLE
feat(BeanstalkdSender): implement beanstalkd sender and perform enque…

### PR DESCRIPTION
**Pull request by rayway30419**
…uing event.payload to target which defined in event.control

### CODE REVIEW

File: internal/senders/beanstlakd_sender.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// BeanstalkdSender - Send Event to beanstalkd
+type BeanstalkdSender struct{}
+
+// SendEvent - #BeanstalkdSender SendEvent
+func (c *BeanstalkdSender) SendEvent(e models.IncomingEvent, d models.Destination) error {
+  host := fmt.Sprintf("%s:%s", e.Control["beanstalkd_host"], e.Control["beanstalkd_port"])
+  conn, err := beanstalk.Dial("tcp", host)
+
+  if err != nil {
+    log.Info("Beanstalk create connection failed: ", err)
```
Comment (Seitk) [inline internal/senders/beanstlakd_sender.go]: use logrus with reference fields, check internal/senders/http_sender.go as example

File: internal/senders/beanstlakd_sender.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  log "github.com/sirupsen/logrus"
+)
+
+
+// BeanstalkdSender - Send Event to beanstalkd
+type BeanstalkdSender struct{}
+
+// SendEvent - #BeanstalkdSender SendEvent
+func (c *BeanstalkdSender) SendEvent(e models.IncomingEvent, d models.Destination) error {
+  host := fmt.Sprintf("%s:%s", e.Control["beanstalkd_host"], e.Control["beanstalkd_port"])
```
Comment (Seitk) [inline internal/senders/beanstlakd_sender.go]: I think we can just pass it as a whole from `Control` then

File: internal/senders/beanstlakd_sender.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  models "github.com/shoplineapp/captin/models"
+  beanstalk "github.com/beanstalkd/go-beanstalk"
+  "fmt"
+  "time"
+  log "github.com/sirupsen/logrus"
+)
+
+
+// BeanstalkdSender - Send Event to beanstalkd
+type BeanstalkdSender struct{}
```
Comment (Seitk) [inline internal/senders/beanstlakd_sender.go]: add interface `interfaces.EventSenderInterface`

File: internal/senders/beanstlakd_sender.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    log.Info("Beanstalk create connection failed: ", err)
+    panic(err)
+  }
+
+  tubeName := fmt.Sprintf("%s.%s", e.Control["tube_namespace"], e.Control["queue_name"])
+  conn.Tube = beanstalk.Tube { Conn: conn, Name: tubeName }
+  jobBody, err := json.Marshal(e.Payload)
+  if err != nil {
+    log.Info(fmt.Sprintf("* Beanstalkd job payload format invalid: %s", string(jobBody)))
+    panic(err)
```
Comment (Seitk) [inline internal/senders/beanstlakd_sender.go]: i think the `dispatcher.sendEvent` didn't recover from panic, probably need to update that method if you want to use panic as function break here

File: core/captin.go [hunk 5]
```diff
@@ -40,6 +40,7 @@ func NewCaptin(configMap interfaces.ConfigMapperInterface) *Captin {
 	store := stores.NewMemoryStore()
 	senderMapping := map[string]interfaces.EventSenderInterface{
 		"http": &senders.HTTPEventSender{},
+		"beanstalkd": &senders.BeanstalkdSender{},
```
Comment (Seitk) [inline core/captin.go]: i think we don't have to add `beanstalkd` as one of the default sender

Comment (rayway30419) [reply core/captin.go]: then we must create senderMapping in event center?

File: senders/beanstlakd_sender.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+var bLogger = log.WithFields(log.Fields{"class": "BeanstalkdSender"})
+
+// BeanstalkdSender - Send Event to beanstalkd
+type BeanstalkdSender struct {
+  interfaces.EventSenderInterface
+}
+
+// SendEvent - #BeanstalkdSender SendEvent
+func (c *BeanstalkdSender) SendEvent(e models.IncomingEvent, d models.Destination) error {
+  host := fmt.Sprintf("%v", e.Control["beanstalkd_host"])
```
Comment (darkcl) [inline senders/beanstlakd_sender.go]: Can use defer, recover and panic for error handling 🤔
https://blog.golang.org/defer-panic-and-recover 

```go
defer func() {
  if err := recover(); err != nil {
    bLogger.WithFields(log.Fields{
      "error": err,
    }).Error(err)
  }
  return err
}()
```

File: senders/beanstlakd_sender.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  conn, err := beanstalk.Dial("tcp", host)
+  if err != nil {
+    bLogger.WithFields(log.Fields{
```
Comment (darkcl) [inline senders/beanstlakd_sender.go]: and panic when receive an error

```go
panic("Beanstalk create connection failed.")
```

**Pull request merged**